File: DataAccess/src/main/java/com/TitleCounter/DataAccess/dto/GameEntryDto.java
package com.TitleCounter.DataAccess.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.*;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GameEntryDto {
    private Long id;

    @Size(max=255)
    private String note;

    @Min(0)
    @Max(10)
    private Long score;

    @NotBlank
    private String status;

    @JsonProperty(value = "date_completed")
    private LocalDate dateCompleted;

    @Min(0)
    private Long time;

    private String platform;

    @NotNull
    @JsonProperty(value = "client_id")
    private Long userId;

    @NotNull
    @JsonProperty(value = "game_id")
    private Long gameId;
}

File: WinFormsApplication/Dto/GameEntryDto.cs
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace hltb.Dto
{
    public class GameEntryDto
    {
        private long id;
        private string note;
        private long score;
        private string status;
        private string dateCompleted;
        private long time;
        private string platform;
        [JsonPropertyName ("client_id")]
        private long userId;
        [JsonPropertyName  ("game_id")]
        private long gameId;

        [JsonPropertyName ("game")]
        private GameDto gameDto;
    }
}
